src/middlewared/middlewared/plugins/apps/compose_utils.py

```diff
@@ -0,0 +1,38 @@
+import itertools
+import typing
+
+from middlewared.service_exception import CallError
+
+from .ix_apps.lifecycle import get_rendered_templates_of_app
+from .utils import PROJECT_PREFIX, run
+
+
+def compose_action(
+    app_name: str, app_version: str, action: typing.Literal['up', 'down'], *,
+    force_recreate: bool = False, remove_orphans: bool = False,
+):
+    compose_files = list(itertools.chain(
+        *[('-f', item) for item in get_rendered_templates_of_app(app_name, app_version)]
+    ))
+    if not compose_files:
+        raise CallError(f'No compose files found for app {app_name!r}')
+
+    args = ['-p', f'{PROJECT_PREFIX}{app_name}', action]
+
+    if action == 'up':
+        args.append('-d')
+        if force_recreate:
+            args.append('--force-recreate')
+        if remove_orphans:
+            args.append('--remove-orphans')
+    elif action == 'down':
+        if remove_orphans:
+            args.append('--remove-orphans')
+        args.append('-v')
+    else:
+        raise CallError(f'Invalid action {action!r} for app {app_name!r}')
+
+    # TODO: We will likely have a configurable timeout on this end
+    cp = run(['docker-compose'] + compose_files + args, timeout=1200)
```

Comment:
I've been thinking about this, maybe we can do this in 2 steps.
1. docker compose pull (using the configured timeout)
2. docker compose [action], with a much much smaller timeout.

Does it make sense to do that? 
We/User would at least have a better visibility what is taking long.
And also will not block for an insane amount of time on docker compose [action].